pkg/apis/core/validation/validation.go

```diff
@@ -1452,6 +1452,33 @@ func validateCSIPersistentVolumeSource(csi *core.CSIPersistentVolumeSource, fldP
 		allErrs = append(allErrs, field.Required(fldPath.Child("volumeHandle"), ""))
 	}
 
+	if csi.ControllerPublishSecretRef != nil {
```

Comment:
for all of these, ensure name is a valid secret name and namespace is a valid namespace name

Reply:
done